app/javascript/react/components/NewMovieForm.js

```diff
@@ -0,0 +1,56 @@
+import React, {useState} from "react"
+import NewMovieFormTile from "./NewMovieFormTile"
+import { Redirect } from "react-router-dom"
+import ApiErrorList from "./ApiErrorList"
+
+const NewMovieForm = (props) => {
+  const [movie, setMovie] = useState({})
+  const [errors, setErrors] = useState([])
+
+  const addMovie = async (formPayload) => {
+    try {
+      const response = await fetch("/api/v1/movies", {
+        credentials: "same-origin",
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "Accept": "application/json"
+        },
+        body: JSON.stringify(formPayload)
+      })
+      if(!response.ok){
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error)
+      }
+      const responseBody = await response.json()
+      if(responseBody.movie){
+        setMovie(responseBody.movie) 
+      } else {
+        setErrors(responseBody.errors)
+      }
+
+    } catch(error) {
+      console.log(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  let fetchReturn
+  if(movie.id) {
+    fetchReturn = <Redirect to={`/movies/${movie.id}`}/>
+  } 
```

Comment:
let's remove the `let` and variable on line 38 - the `<Redirect />` does not need to be included in the rendered JSX below. When the Redirect line is evaluated it will issue the Redirect:
```js
if (movie.id) {
  <Redirect to={} />
}
```